experimental/design-notes.txt

```diff
@@ -0,0 +1,112 @@
+abstraction (MeterProvider->Meter->Instrument->BoundInstrument)?
+
+3. MM landed on a different set of instrument types than OT did. It would be good to understand why we believe OT's instrument set is the right set to pick?
+MM has Timer, Counter, Gauge, DistributionSummary, LongTaskTimer, FunctionCounter, FunctionTimer, and TimeGauge. I think the conceptual match up is roughly:
+Timer                   NA
+Counter                 Counter
+DistributionSummary     ValueRecorder
+Gauge                   ValueObserver
+LongTaskTimer           NA
+FunctionCounter         SumObserver
+FunctionTimer           NA
+TimeGauge               NA
+NA                      UpDownCounter
+NA                      UpDownSumObserver
+
+
+4. MM has a different model for observation/asynchronicity than OT. MM has the Meter (in MM terminology) provide a callback function that when invoked returns
+the value of the measurement for that measurement interval. OT also uses a callback but the callback is expect to invoke a Observe() API to record potentially
+many data points and has void return type. I think the additional complexity in the OT model is a side-effect of the design decision to have unbound instruments
+that might need to set the value of potentially many LabelSets whereas in MM all Meters (in MM terminology) are bound to a single set of labels and will
+have exactly 1 data point per time period. Does the value of unbound instruments justify this additional complexity?
+
+5. MM has no batch mode for logging metrics, how valuable of a feature is this given its additional complexity to implement? Is it purely a performance optimization
+or is there a scenario where the atomicity of the measurements matter?
+
+6. MM has a lot of API surface dedicated to configuring Meters (in MM terminology) and tags. I am guessing this is the kind of functionality that would be sitting in OT's
+Views API?
+
+
+IfxMetricsExtensions:
+
+Is there a better doc than https://genevamondocs.azurewebsites.net/collect/instrument/ifx/ifxextensions.html? This page seems to assume I am already pretty familiar
+with Geneva metrics and only describes some specific addendums.
+
+Prometheus.NET - https://github.com/prometheus-net/prometheus-net
+
+1. Naming - What MM calls Meters and OT calls bound instruments, Prometheus calls Metrics.
+2. Similar to MM and different from OT, metrics can be created directly from static functions on a global Metrics class.
+3. Prometheus also has different metric types (instrument types in OT terminology). It has Counter, Guage, Summary, and Histogram. The best mapping appears to be:
+Counter       Counter
+Gauge         UpDownCounter or ValueRecorder?
+Summary       ValueRecorder
+Histogram     ValueRecorder
+NA            *Observer
+
+4. Prometheus uses Observe() API where OT uses Record() API. Prometheus doesn't appear to have asynchronous/callback style APIs.
```

Comment:
I think we do need an "async" callback for reporting data.  Or at least some examples of how to do this.  Need to resolve the issue of Sum vs Incremental dataset.  Cause any "Observed" value implies a Time Range, and thus, some kind of aggregation.  This needs to be reviewed a bit deeper I think.